### DISCUSSION TITLE
[FEAT] Genomic contexts

### DISCUSSION THREAD

**Discussion by jananiravi** [JRaviLab/MolEvolvR#23] [Ideas]
# Genomic context aka neighborhood analyses
- [ ] `edirect` and `eutils` to fetch genes of interest (from a list)
   - [ ] + slurm
- [ ] user-input: no. of neighbors on each side (default: ±7)
- [ ] user-input: filter by intergenic distance (200bp)
- [ ] identify orientation of genes
- [ ] back to tab-delim format with `<- || ->` | `A -> B -> C||<- D`
- [ ] option: indicate query with * or some other marker
- [ ] first: append final GC column to the BLAST output file
- [ ] add DA for each of the neighbors' accessions
- [ ] finally: replace with GC column above with DA+GC `A+B -> C+D`

## Existing tools/papers
- [ ] [NCBI Genome ProtMap](https://www.ncbi.nlm.nih.gov/genomes/static/proteinclusters_help.html#protmap)
- [ ] `rentrez`: [CRAN tutorial](https://cran.r-project.org/web/packages/rentrez/vignettes/rentrez_tutorial.html)
- [ ] Is there a way to retain intergenic distance | to use later as edge weight in networks | Check 2020 PLoSCompBio pipeline paper
- [ ] MiST DB | 2020 paper mentions this | 2024 version doesn't!
- [ ] COGNAT: a web server for comparative analysis of genomic neighborhoods | Klimchuk et al., Biol. Direct 2020
- [ ] Systematic prediction of functionally linked genes in bacterial and archaeal genomes | Shmakov et al., Nat Protocols 2020
- [ ] Gene Graphics | Harrison et al., Bioinformatics 2018
- [ ] Conserved Genomic Neighborhood ... | Esch et al., BMC Bioinformatics 2020
- [ ] `gggenomes`: https://thackl.github.io/gggenomes
- [ ] `GeCoViz`: https://gecoviz.cgmlab.org | [paper](https://academic.oup.com/nar/article/50/W1/W352/6593547)

### Predicting operons
- [ ] [Operon-mapper](https://biocomputo.ibt.unam.mx/operon_mapper/) | 2018, Bioinformatics, Merino Lab
- [ ] [MicrobesOnline Operon predictions](http://www.microbesonline.org/operons/) | 2005, NAR, Price, ... Arkin
- [ ] [OperonHunter, using visual representation learning](https://www.nature.com/articles/s41598-021-81169-9) | 2021, Sci Reports, Assaf, ... Stevens (Rick Stevens, BV-BRC PI)

- [ ] @epbrenner: please add other papers from your list too

**Comment by jananiravi** [comment]
@epbrenner, here's your new companion [e-book on eutils, entrez](https://www.ncbi.nlm.nih.gov/books/NBK179288/), and more! (Related: [rentrez](https://cran.r-project.org/web/packages/rentrez/vignettes/rentrez_tutorial.html))

**Comment by daud99** [comment]
@jananiravi Can you assigned this task to me. I'm an Outreachy applicant selected for this session?

**Comment by jananiravi** [comment]
That's excellent, @daud99! Welcome aboard! Are you an outreachy applicant, too? If so, please email me & I can add you to Slack.

I converted this to discussions before noticing your comment -- so, I cannot assign you. But please keep the discussion going & get started on your PR. How do you plan to broach this? @epbrenner will be your primary contact (starting next week).

**Reply by daud99** [reply]
done @jananiravi

**Comment by jananiravi** [comment]
@epbrenner, edited the original note and added an _operon prediction_ section above, too. Hopefully, in addition to laying out genomic neighborhood, this provides additional context.
